src/common/string/compare.ts

```diff
@@ -45,3 +45,22 @@ export const caseInsensitiveStringCompare = (
 
   return fallbackStringCompare(a.toLowerCase(), b.toLowerCase());
 };
+
+export const orderCompare = (order: string[]) => (a: string, b: string) => {
+  const idxA = order.indexOf(a);
+  const idxB = order.indexOf(b);
+
+  if (idxA === idxB) {
+    return 0;
```

Comment:
Shouldnt we compare the strings if the index of the order is the same? Like fallback to alphabetically?

Reply:
By putting 0, the order will stay the same as the initial list. We must not order it alphabetically because it's doesn't make sense when it's an ID.